CONTRIBUTING.md

```diff
@@ -0,0 +1,368 @@
+  and mutability;
+* Add documentation to anything that says `public` or `protected`. The
+  checkstyle doesn't require documentation on package-local or `private` fields, but
+  that might be a good idea, too, if you're doing some interesting/non-trivial
+  work in those places.
+
+These are just some general guidelines, but
+[Checkstyle](https://github.com/checkstyle/checkstyle/blob/master/README.md) and
+[Android Lint](https://developer.android.com/studio/write/lint#overview)
+are the ultimate authorities on this, and will tell you if something is
+badly formatted, or exciting a known shortcoming in Java, Android, or
+their library ecosystem.
+
+#### Build and Validate Your Work
+
+This will perform a clean build, run Checkstyle, Android Lint, and all unit
+tests. This must complete successfully before proposing a PR.
+
+```
+./gradlew clean build
+```
+
+#### Run Instrumentation Tests
+
+The instrumentation tests presume the presence of various backend resources.
+Currently, there is no mechanism for contributors to easily allocate
+these resources. 
+
+AWS maintainers can gain access to `awsconfiguration.json` and
+`testconfiguration.json` files in an S3 bucket, to find
+configurations suitable for running the integration tests.
+
+The file locations can be found in the `Configure test information` command in the `.circleci/config.yml` file.
+
+To run a __specific__ test:
+
+```bash
+test='com.amazonaws.mobileconnectors.cognitoidentityprovider.SignInPersistenceIntegrationTest#testSignIn'
+./gradlew cAT -Pandroid.testInstrumentationRunnerArguments.class="$test"
+```
+
+To run __all__ tests:
+
+```
+./gradlew cAT
+```
+
+## Reporting Bugs/Feature Requests
+
+We welcome you to use the GitHub issue tracker to report bugs or suggest
+features.
+
+When filing an issue, please check [existing open](https://github.com/aws-amplify/aws-sdk-android/issues)
+and [recently closed](https://github.com/aws-amplify/aws-sdk-android/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20)
+issues to make sure somebody else hasn't already reported the issue.
+Please try to include as much information as you can. Details like these
+are useful:
+
+* The version of the Framework you are using
+* Details and configurations for any backend resources that are relevant
+* A full exception trace of an error you observe
+* A statement about what system behavior you _expect_, alongside the
+  behavior you actually observe
+
+## Contributing via Pull Requests
+
+This is mostly the same as [GitHub's guide on creating a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request).
+
+First, create a _fork_ of `aws-sdk-android`. Clone it, and make changes to this _fork_.
+
+```
+git clone git@github.com:your_username/aws-sdk-android.git 
+```
+
+After you have tested your feature/fix, by adding sufficient test coverage, and
+validating Checkstyle, lint, and the existing test suites, you're ready to
```

Comment:
`validating Checkstyle, lint, and the existing test suites` -->

`validating Android Lint and the existing test suites,`

Reply:
done